commands/retrieval_client_daemon_test.go

```diff
@@ -0,0 +1,65 @@
+package commands_test
+
+import (
+	"bytes"
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ipfs/go-ipfs-files"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/tools/fast"
+	"github.com/filecoin-project/go-filecoin/tools/fast/fastesting"
+	"github.com/filecoin-project/go-filecoin/tools/fast/series"
+)
+
+func TestSelfDialRetrievalGoodError(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	assert := assert.New(t)
+	require := require.New(t)
+
+	ctx := context.Background()
+
+	ctx, env := fastesting.NewTestEnvironment(ctx, t, fast.EnvironmentOpts{})
+	// Teardown after test ends.
+	defer func() {
+		err := env.Teardown(ctx)
+		require.NoError(err)
+	}()
+
+	// Update genesis miner's peerid
+	var minerAddr address.Address
+	err := env.GenesisMiner.ConfigGet(ctx, "mining.minerAddress", &minerAddr)
+	require.NoError(err)
+	details, err := env.GenesisMiner.ID(ctx)
+	require.NoError(err)
+	msgCid, err := env.GenesisMiner.MinerUpdatePeerid(ctx, minerAddr, details.ID, fast.AOPrice(big.NewFloat(1.0)), fast.AOLimit(300))
+	require.NoError(err)
```

Comment:
All of this should already have happened. The environment setups the `GenesisMiner` by calling  `SetupGenesisNode` which calls `MinerUpdatePeerid`. Where you running to some issue or just added this out of habit?

https://github.com/filecoin-project/go-filecoin/blob/master/tools/fast/fastesting/basic.go#L69
https://github.com/filecoin-project/go-filecoin/blob/master/tools/fast/series/setup_genesis_node.go#L39

Reply:
I ran into an issue.  The miner actor and genesis pid differed.

Reply:
I'll toy with this a bit and see if I can find the issue. We shouldn't have to do this.